examples/cooperative_preempt.rs

```diff
@@ -18,40 +18,40 @@ fn main() {
     //
     // There are two ways of yielding control:
     //
-    //  * Local::yield_if_needed(), which will yield if the task has run for too
-    //    long. What "too long" means is an implementation detail, but it will be
-    //    always somehow related to the latency guarantees that the task queues want
-    //    to uphold in their `Latency::Matters` parameter (or
+    //  * glommio::executor().yield_if_needed(), which will yield if the task has
```

Comment:
probably best if `yield_if_needed` is totally free-floating, and not pegged to the executor.
It's possible to optimize it further so that not even an indirection to the executor is necessary (seastar, for example, just saves the address of the yield variables in a TLS variable). While their yield is a bit different, we could in principle use a similar optimization

Reply:
`yield_if_needed` is one of those that fits everywhere. It fits being in the `Task` itself because we're yielding a task. But it also fits being in the executor as it relies on the internal state to make a decision. 

I could do as I did for `local` and `local_into`, i.e. make them part of the executor struct but also add a free function alias for convenience. Let me know what feels right to you!

Reply:
I agree that `yield_if_needed` can be accessed through many places as a concept, I am just concerned about the optimization opportunities that may be lost, but maybe that's a different conversation.

Reply:
I added a free function `yield_if_needed`, I think it makes sense to make it easily accessible just like the spawn functions :+1: